scripts/mk.js

```diff
@@ -11805,21 +11821,34 @@ function getHoleSegmentDist(currentRes, x,y, x1,y1, u1,v1) {
 	return currentRes;
 }
 
-function objet(iX, iY) {
-	var lMap = getCurrentLMap(collisionLap);
+function checkCollision(kX, kY, oldX, oldY, objX, objY, hitboxSize) {
+    for (var i = 0; i <= collisionSubChecks; i++) {
```

Comment:
Not a fan of this solution which adds lots of extra calculation and is still not 100% exact. I created a function called `pointCrossRectangle` which in addition to `pointInRectangle` allows to perform an exact collision check. You have a usage example here => https://github.com/tmalahie/mkpc/blob/5a42cd317c452df97344d191bb304c6c95176b76/scripts/mk.js#L11970